src/main/java/org/quiltmc/loader/impl/entrypoint/GamePatch.java

```diff
@@ -128,12 +129,18 @@ protected boolean isPublicInstance(int access) {
 		return ((access & 0x0F) == (Opcodes.ACC_PUBLIC | 0 /* non-static */));
 	}
 
-	public void process(QuiltLauncher launcher, Function<String, ClassReader> classSource, Consumer<ClassNode> classEmitter) {
+	public void process(QuiltLauncher launcher, String namespace, Function<String, ClassReader> classSource, Consumer<ClassNode> classEmitter) {
 		throw new AbstractMethodError(getClass() + " must override one of the 'process' methods!");
 	}
+	public void process(QuiltLauncher launcher, Function<String, ClassReader> classSource, Consumer<ClassNode> classEmitter) {
+		process(launcher, QuiltLoaderImpl.INSTANCE.tryGetGameProvider().getNamespace(), classSource, classEmitter);
+	}
 
+	public void process(QuiltLauncher launcher, String namespace, GamePatchContext context) {
+		process(launcher, namespace, context::getClassSourceReader, context::addPatchedClass);
+	}
 	public void process(QuiltLauncher launcher, GamePatchContext context) {
-		process(launcher, context::getClassSourceReader, context::addPatchedClass);
+		process(launcher, QuiltLoaderImpl.INSTANCE.tryGetGameProvider().getNamespace(), context);
```

Comment:
`tryGetGameProvider()` should be replaced with `getGameProvider()` (it has a better error message if it's not set).